### ISSUE TITLE
0.35+ regression - deps of dead functions are not shaken

### ISSUE THREAD

**Issue by nolanlawson** [rollup/rollup#950]
Sorry for opening an issue without a small reproducible test case. It's reproducible, but it's not small.

We're still using Rollup 0.34.13 in PouchDB. Most of the major issues in 0.35.x seem to be fixed, but there's one remaining (https://github.com/pouchdb/pouchdb/pull/5689), which is that our coverage tests are now failing because of additional code that is not being tree-shaken.

It seems to be a deconflicting issue. We have several functions called `preprocessAttachments`, one of which is dead code. This dead one calls `preprocessAttachment` (among other functions), which is being left in the codebase even though `preprocessAttachments` is gone.

STR:

```
git clone --depth 1 --single-branch \
  --branch greenkeeper-rollup-0.35.11 https://github.com/pouchdb/pouchdb.git
cd pouchdb
npm install
cat packages/node_modules/pouchdb/lib/index.js
```

Here is the [before](https://gist.githubusercontent.com/nolanlawson/0dd3813093bcdf8f3414452743daaad7/raw/e2d86323a5ecca4e8a33b408bc9bb03c5d6e395c/pouchdb-before.js) and [after](https://gist.githubusercontent.com/nolanlawson/0dd3813093bcdf8f3414452743daaad7/raw/e2d86323a5ecca4e8a33b408bc9bb03c5d6e395c/pouchdb-after.js). You can diff those two files and see the dead code that was included:

![screenshot 2016-09-16 22 02 13](https://cloud.githubusercontent.com/assets/283842/18606069/4ab0dd6e-7c59-11e6-8601-fa4c770c0701.png)

Sorry for the lack of a small test case, but I figured it was better to file an issue early, since I at least have a repro.

**Comment by nolanlawson** [comment]
OK, I have found a minimal repro for this bug. Compare:

- [old version (0.34.13)](http://rollupjs.org/?version=0.34.13&shareable=JTdCJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm1vZHVsZU5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwJTdCJTIwdXNlZEZ1bmMlMjAlN0QlMjBmcm9tJTIwJy4lMkZtb2R1bGUuanMnJTNCJTVDbnVzZWRGdW5jKCklMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIybW9kdWxlLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmZ1bmN0aW9uJTIwZGVwT2ZVbnVzZWRGdW5jKCklMjAlN0IlNUNuJTIwJTIwY29uc29sZS5sb2coJ3VudXNlZCUyMGRlcCEnKSU1Q24lN0QlNUNuZnVuY3Rpb24lMjB1bnVzZWRGdW5jJTIwKCklMjAlN0IlNUNuJTIwJTIwZGVwT2ZVbnVzZWRGdW5jKCklM0IlNUNuJTIwJTIwY29uc29sZS5sb2coJ3VudXNlZCEnKSU1Q24lN0QlNUNuZnVuY3Rpb24lMjB1c2VkRnVuYyUyMCgpJTIwJTdCJTVDbiUyMCUyMGNvbnNvbGUubG9nKCd1c2VkIScpJTVDbiU3RCU1Q25leHBvcnQlMjAlN0IlMjB1c2VkRnVuYyUyQyUyMHVudXNlZEZ1bmMlMjAlN0QlMjIlN0QlNUQlN0Q=)
- [new version (0.36.1)](http://rollupjs.org/?version=0.36.1&shareable=JTdCJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm1vZHVsZU5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwJTdCJTIwdXNlZEZ1bmMlMjAlN0QlMjBmcm9tJTIwJy4lMkZtb2R1bGUuanMnJTNCJTVDbnVzZWRGdW5jKCklMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIybW9kdWxlLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmZ1bmN0aW9uJTIwZGVwT2ZVbnVzZWRGdW5jKCklMjAlN0IlNUNuJTIwJTIwY29uc29sZS5sb2coJ3VudXNlZCUyMGRlcCEnKSU1Q24lN0QlNUNuZnVuY3Rpb24lMjB1bnVzZWRGdW5jJTIwKCklMjAlN0IlNUNuJTIwJTIwZGVwT2ZVbnVzZWRGdW5jKCklM0IlNUNuJTIwJTIwY29uc29sZS5sb2coJ3VudXNlZCEnKSU1Q24lN0QlNUNuZnVuY3Rpb24lMjB1c2VkRnVuYyUyMCgpJTIwJTdCJTVDbiUyMCUyMGNvbnNvbGUubG9nKCd1c2VkIScpJTVDbiU3RCU1Q25leHBvcnQlMjAlN0IlMjB1c2VkRnVuYyUyQyUyMHVudXNlZEZ1bmMlMjAlN0QlMjIlN0QlNUQlN0Q=)

In the new version, `depOfUnusedFunc` is (incorrectly) included in the final bundle.

![2016-09-29 12_56_44-rollup js - microsoft edge](https://cloud.githubusercontent.com/assets/283842/18970008/6dc3715a-8644-11e6-8acd-2a41f624d086.png)

**Comment by kzc** [comment]
Dupe of #977

**Comment by Rich-Harris** [comment]
Fixed since 0.37

**Comment by nolanlawson** [comment]
Awesome! We just got a full test pass (https://github.com/pouchdb/pouchdb/pull/6041), meaning code coverage is passing because the dead functions were shaken out. We're now able to upgrade from 0.34.13 to 0.37.1. So happy about this.
